ArmPkg/Drivers/TimerDxe/TimerDxe.c

```diff
@@ -143,7 +143,7 @@ TimerDriverSetTimerPeriod (
     // mTimerTicks = TimerPeriod in 1ms unit x Frequency.10^-3
     //             = TimerPeriod.10^-4 x Frequency.10^-3
     //             = (TimerPeriod x Frequency) x 10^-7
-    TimerTicks = MultU64x32 (TimerPeriod, ArmGenericTimerGetTimerFreq ());
+    TimerTicks = MultU64x64 (TimerPeriod, ArmGenericTimerGetTimerFreq ());

```

Comment:
The change makes sense.
An alternative change would be to trust the compiler and just do the straight multiplication.
To my knowledge, this has never caused an issue outside of very ancient visual studio versions on x86, and this code is Arm-specific.

Reply:
Thanks for the information. The change is updated.